### PR TITLE
DS-4405: Deduplicate value attribute labels

### PR DESCRIPTION
**Pull request by mwm-displayr**
* Remove check for duplicate value attribute labels from isValueAttributesMergable
* Add function deduplicateValAttrNames which is called in stackedValueAttributes to append the value attribute value to the label if the label is duplicated

### CODE REVIEW

File: R/stacking.R [hunk 1]
```diff
@@ -786,8 +789,7 @@ stackingSpecifiedByVariable <- function(manual.stacking,
             !isValueAttributesMergable(v.val.attr[removeNA(group.ind)]))
         {
             warning("The manual stacking input '", input.text,
-                    "' has been ignored as it contains variables with mismatching types or value attributes. ",
-                   "Every value must have identical labels in each variable it appears in.")
+                    "' has been ignored as it contains variables with mismatching types or value attributes.")
```
Comment (chrisfacer) [inline R/stacking.R:792]: Did you mean to remove the new addition to this comment?

File: tests/testthat/test-stacking.R [hunk 2]
```diff
@@ -728,3 +728,22 @@ if (file.exists("cola19 stacked.sav"))
 
 if (file.exists("cola20 stacked.sav"))
     file.remove("cola20 stacked.sav")
+
+test_that("DS-4405: Stacking with duplicate labels",
+{
+    expect_equal(deduplicateValAttrNames(list(Dislike = 0, Dislike = 1, Hate = 2)),
+                 list("Dislike [Value: 0]" = 0, "Dislike [Value: 1]" = 1, Hate = 2))
+    input.file <- findInstDirFile("testdata/ColaTracking-bugdata.sav")
```
Comment (chrisfacer) [inline tests/testthat/test-stacking.R:736]: "testdata" not needed here I think. Only works locally when I remove it and just have the file name.

File: R/stacking.R [hunk 3]
```diff
@@ -963,12 +965,12 @@ isValueAttributesMergable <- function(v.val.attrs)
 
     # Check that there aren't multiple values per name
```
Comment (chrisfacer) [inline R/stacking.R:966]: Can remove this whole block I think.

Comment (chrisfacer) [reply R/stacking.R:966]: inlcuding unique.names

File: R/stacking.R (hunk 1 above)
Comment (mwm-displayr) [reply R/stacking.R:792]: No. Thanks for catching that!

File: tests/testthat/test-stacking.R (hunk 2 above)
Comment (mwm-displayr) [reply tests/testthat/test-stacking.R:736]: Makes sense now seeing the code.

**Review by chrisfacer — APPROVED** [review]
Looks good to me.

**Comment by mwm-displayr** [comment]
@chrisfacer seems like you were unsure yesterday if users would like the new behaviour of updating the labels, so I added a warning when this occurs. Seems worthwhile in the situation covered by this test https://github.com/Displayr/flipData/pull/61/files#diff-ef534d52b6d2ad762c8c2a0c19312c123002e007b1a8e894c2cfa329b485522fL680. The user could edit the values to match in both variables and rerun. What do you think?

**Comment by chrisfacer** [comment]
Yes @mwm-displayr I think it helps them to see the problem and either fix it in the source files, or to know they they might need to do a quick label fix or category merge once they have imported their new file.

**Review by chrisfacer — APPROVED** [review]

**Pull request merged**